adobe/react-spectrum
# Fix backspace to shift focus to previous element in useDateSegment

Closes https://github.com/adobe/react-spectrum/issues/5261 , https://github.com/adobe/react-spectrum/issues/5223

## ✅ Pull Request Checklist:


## 📝 Test Instructions:



### Test 1: Re-entering Data in the Time Field

1. Visit the [Date Field Example in Storybook](http://localhost:9003/?path=/story/react-aria-components--date-field-example&providerSwitcher-express=false&strict=true).
2. Enter "45" into the Number Field for minutes.
3. Clear the entry using the backspace key.
4. Enter a new value, such as "30", for minutes.
5. Confirm that you can correctly re-enter the minutes.

### Test 2: Backspace Focus Shift on Empty DateSegment

1. Visit the [Date Field Example in Storybook](http://localhost:9003/?path=/story/react-aria-components--date-field-example&providerSwitcher-express=false&strict=true).
2. Enter "45" into the Number Field for minutes.
3. Use the backspace key to clear the minutes entered.
4. Press backspace again when the field is empty (showing placeholder).
5. Confirm that the focus shifts to the previous field.
6. Refocus on the minutes field and enter a new value, such as "30".
7. Confirm that the minutes can be correctly re-entered.

## 🧢 Your Project:
